### PR TITLE
Adds support for parsing arrays

### PR DESCRIPTION
**Pull request by theblixguy**
This PR adds support for parsing an array from the Config plist. It probably will not work recursively (array inside array...), but I can fix that if needed.

It's a bit hacky - the value is returned as a variadic tuple, however there's no way to cast the value as a variadic tuple, so you have to do a bit of string manipulation to trim off some parts. But hey, it works!

### CODE REVIEW

File: configen/FileGenerator.swift [hunk 1]
```diff
@@ -141,4 +159,14 @@ extension String {
   var trimmed: String {
     return (self as NSString).trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
   }
+
+  func match(regex: String) -> Bool {
+    do {
+      let regex = try NSRegularExpression(pattern: regex, options: NSRegularExpression.Options.caseInsensitive)
```
Comment (KaneCheshire) [inline configen/FileGenerator.swift]: Could probs just do `options: .caseInsensitive)` here?

File: configen/FileGenerator.swift [hunk 2]
```diff
@@ -131,6 +137,18 @@ struct FileGenerator {
...
+    var rawTypeCopy = rawType
+    var rawValueStr = "\(rawValue)"
+
+    rawTypeCopy = String(rawTypeCopy.dropFirst()) // drop [
+    rawTypeCopy = String(rawTypeCopy.dropLast()) // drop ]
+
+    rawValueStr = String(rawValueStr.dropFirst()) // drop (
+    rawValueStr = String(rawValueStr.dropLast()) // drop )
+
+    return "Array<\(rawTypeCopy)>(arrayLiteral: \(rawValueStr))"
```
Comment (KaneCheshire) [inline configen/FileGenerator.swift]: How will this output in the generated file, will it literally be `Array<String>(arrayLiteral: ...)`? Or will it be the nicer `[...]` style?

File: configen/FileGenerator.swift (hunk 1 above)
Comment (theblixguy) [reply configen/FileGenerator.swift]: Done 👍

File: configen/FileGenerator.swift (hunk 2 above)
Comment (theblixguy) [reply configen/FileGenerator.swift]: Yeah, fixed now! 👍

**Review by KaneCheshire — APPROVED** [review]

**Pull request merged**